### PR TITLE
Fix query builder scroll

### PR DESCRIPTION
**Pull request by sapansanu**
###### Before submitting the PR, please make sure you do the following 
-  [ ] If there are changes to the backend codebase, run the backend tests with `lein test && lein eastwood && lein bikeshed && lein docstring-checker && ./bin/reflection-linter`
-  [ ] Run the frontend and integration tests with  `yarn && yarn run prettier && yarn run lint && yarn run flow && ./bin/build version uberjar && yarn run test`)
-  [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
(unless it's a tiny documentation change).

### CODE REVIEW

**Comment by sapansanu** [comment]
## Query Builder - View and GroupBy are scrollable / FilterBy elements are contained within query builder container

I have made the view and groupby sections scrollable.

The filterby section which was overflowing outside the query builder container is now contained within.

The selection model for filterby opens to the left side and does not overflow from page width.

TODO:
The '+' button for View and GroupBy sections is not static right now. I need suggestions on how to make the '+' button for View and groupBy sections dynamic like FilterBy section.

image

File: frontend/src/metabase/query_builder/components/GuiQueryEditor.jsx [hunk 1]
```diff
@@ -428,22 +430,23 @@ export default class GuiQueryEditor extends Component {
...
         </div>
         <div className="GuiBuilder-row flex flex-full">
-          {this.renderViewSection()}
+          {this.renderFilterSection()}
+          {this.renderViewSection()}          
           {this.renderGroupedBySection()}
           <div className="flex-full" />
           {this.props.children}
-          <ExtendedOptions {...this.props} />
+          {/* <ExtendedOptions {...this.props} /> */}
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/GuiQueryEditor.jsx]: This definitely shouldn't be commented out.

**Comment by tlrobinson** [comment]
Thanks for working on this, however this changes the layout of the query builder bar pretty significantly, such that each section always takes an equal amount of space:

screenshot 2018-05-08 11 22 42

It also disables the "extended options" section, which we obviously don't want to do.

It looks like at some point the 2 row layout for narrow screens broke, which makes the lack of scrolling worse.

I think @mazameli and/or @kdoh should probably chime in with a better thought out solution.

**Review by tlrobinson — CHANGES_REQUESTED** [review]
* Need design direction from @mazameli and/or @kdoh 
* Shouldn't have each section take equal amount of space
* Should probably fix the broken 2 row layout
* Shouldn't disable extended options

**Comment by kdoh** [comment]
I'm going to go ahead and close this. We're going to be making some changes soon that should remove the need for this less than ideal UX situation in the first place, and I don't think the specific implementation here is how we'd want to handle this regardless. Thanks for the effort @sapansanu.

**Pull request closed without merging**